src/Storages/MergeTree/ReplicatedMergeTreeLogEntry.cpp

```diff
@@ -184,6 +212,21 @@ void ReplicatedMergeTreeLogEntryData::readText(ReadBuffer & in)
                 }
                 else if (checkString("into_uuid: ", in))
                     in >> new_part_uuid;
+                else if (checkString("deduplicate_by_columns: [", in))
+                {
+                    Strings new_deduplicate_by_columns;
+                    for (;;)
+                    {
+                        String tmp_column_name;
+                        readJSONString(tmp_column_name, in);
```

Comment:
This is quite dangerous, because JSON does not support non-unicode data while we support arbitrary bytes for column names.

Reply:
This will lead to idiosynchrasy as two different escaping methods will be used in the same file.
Let's reuse existing "tsv-escaped" method.

Reply:
Well, I've done some research, and it looks like the method we use with `TabSeparatedWithNames` (I believe this is what you meant by 'tsv-secaped') wouldn't cut it:
INPUT as stated in C++ code:
```
{"name with space", "\"column\"", "'column'", "колонка", "\u30ab\u30e9\u30e0", "\x00\x01\x03 column \x10\x11\x12"},
```

writeEscapedString
```
deduplicate_by_columns: [name with space,"column",\'column\',колонка,カラム,\0 column ]
```

writeJSONString
```
deduplicate_by_columns: ["name with space","\"column\"","'column'","колонка","カラム","\u0000\u0001\u0003 column \u0010\u0011\u0012"]
```

Please note how `writeEscapedString` eats all non-unicode binary.

Reply:
> Please note how writeEscapedString and writeCSV eats all non-unicode binary.

No, they don't eat binary data. The bytes are written but they are lost in copy-paste.
Actually our JSON format is also binary safe, but it can be misleading to applications.

Reply:
```
$ clickhouse-client --query "SELECT '\x00\x01\x02\x03'"
\0
$ clickhouse-client --query "SELECT '\x00\x01\x02\x03'" | xxd
00000000: 5c30 0102 030a                           \0....
```

Reply:
Agree, my bad. Changed that to CSV format, since that makes it easier to parse list of columns. Right now serialized list of columns from the unit tests like this:
```
...
deduplicate_by_columns: "name with space","""column""","'column'","колонка","カラム"," column "
...
```

However, I had to ditch using `\0` in test data since it makes impossible to validate result with `regex`.